misaligned/misaligned.cs

```diff
@@ -1,23 +1,53 @@
-﻿using System;
+using System;
+using System.Collections.Generic;
 using System.Diagnostics;
 
-namespace MisalignedSpace {
-    class Misaligned {
-        static int printColorMap() {
-            string[] majorColors = {"White", "Red", "Black", "Yellow", "Violet"};
-            string[] minorColors = {"Blue", "Orange", "Green", "Brown", "Slate"};
-            int i = 0, j = 0;
-            for(i = 0; i < 5; i++) {
-                for(j = 0; j < 5; j++) {
-                    Console.WriteLine("{0} | {1} | {2}", i * 5 + j, majorColors[i], minorColors[i]);
-                }
-            }
-            return i * j;
-        }
-        static void Main(string[] args) {
-            int result = printColorMap();
-            Debug.Assert(result == 25);
-            Console.WriteLine("All is well (maybe!)");
+namespace MisalignedSpace
+{
+  class Misaligned
+  {
+    private static List<string> _majorColorList = new List<string>();
+    private static List<string> _minorColorList = new List<string>();
+    private static List<int> _pairNumber = new List<int>();
+
+    static int PrintColorMap()
+    {
+      string[] majorColors = { "White", "Red", "Black", "Yellow", "Violet" };
+      string[] minorColors = { "Blue", "Orange", "Green", "Brown", "Slate" };
+      int i = 0, j = 0;
+      for (i = 0; i < 5; i++)
+      {
+        for (j = 0; j < 5; j++)
+        {
+          Console.WriteLine("{0} | {1} | {2}", i * 5 + j, majorColors[i], minorColors[i]);
+          _pairNumber.Add(i * 5 + j);
+          _majorColorList.Add(majorColors[i]);
+          _minorColorList.Add(minorColors[i]);
         }
+      }
+      return i * j;
+    }
+    static void Main(string[] args)
+    {
+      int result = PrintColorMap();
+      Debug.Assert(result == 25);
+      Console.WriteLine("Count status - Success");
+
+      //Check the sequence of major and minor colors
+      Debug.Assert(_majorColorList[1] == "White");
+      Debug.Assert(_minorColorList[1] == "Blue");
+      Console.WriteLine("Sequence Check - Success");
+
+      Debug.Assert(_majorColorList[10] == "Red");
+      Debug.Assert(_minorColorList[10] == "Slate");
+      Console.WriteLine("Sequence Check - Success");
+
+      //Check the pair number
+      Debug.Assert(_pairNumber[10] == 11);

```

Comment:
if _pairNumber[10] is 11 , then _majorColorList[1] should be White and _minorColorList[1] should be Orange.
the expectation (as expressed in the test asserts) from the production code is not easy to comprehend. the indexing is ambiguous.

one place zero based indexinig is used and another place 1 based indexing. this needs to be clear.